plugins/experimental/wasm/wasm_main.cc

```diff
@@ -181,11 +189,21 @@ http_event_handler(TSCont contp, TSEvent event, void *data)
   case TS_EVENT_HTTP_READ_CACHE_HDR:
     break;
 
-  case TS_EVENT_HTTP_TXN_CLOSE:
+  case TS_EVENT_HTTP_TXN_CLOSE: {
     context->onDone();
     context->onDelete();
 
-    if (context->wasm() == wasm_config->wasm.get()) {
+    bool found = false;
+    for (auto it = wasm_config->configs.begin(); it != wasm_config->configs.end(); it++) {
+      std::shared_ptr<ats_wasm::Wasm> wbp = it->first;
+      if (wbp.get() == context->wasm()) {
+        found = true;
+      }
+
+      break;
```

Comment:
same thing, this will stop after the first iteration

Reply:
same here. will fix and test more.

Reply:
fixed